src/app/file-browser/components/file-list/file-list.component.ts

```diff
@@ -561,4 +545,16 @@ export class FileListComponent
 			this.visibleItems.delete(event.component);
 		}
 	}
+
+	private initializeFolder() {
```

Comment:
great drying!

One suggestion: this says `initializeFolder` but it seems to be called on both init and changes and it does more than just initialize a single folder.  Is that the right name (maybe it is!).  Is this more of a `syncState` or `

Possibly it could be something like: such as `syncStateWithRoute` which I think captures the intention.